### ISSUE TITLE
Input eaten after using Input.parse_input_event()

### ISSUE THREAD

**Issue by fmenozzi** [godotengine/godot#45043]
**Godot version:**
3.2.3.stable

**OS/device including version:**
OS: Ubuntu 16.04 LTS
Graphics card: Intel Ivybridge Mobile Integrated Graphics
Backend: GLES3

**Issue description:**
I wanted to allow for items in `OptionButton` to automatically grab focus when the button itself is clicked. Because I was unable to find a built-in way to grab focus on individual elements in an `OptionButton`, I decided to try and simulate this by simulating a `ui_down` input event action once the `OptionButton` itself is pressed. The setup I'm using is shown below (this is the same setup as the attached minimum reproduction project):

Scene setup:
![project-setup](https://user-images.githubusercontent.com/4451432/104103860-506fc000-5259-11eb-8d3b-b198b356283b.png)

Script:
```gdscript
extends VBoxContainer

func _ready() -> void:
    $Row1/OptionButton.grab_focus()

    $Row1/OptionButton.connect('pressed', self, '_on_option_button_1_pressed')
    $Row2/OptionButton.connect('pressed', self, '_on_option_button_2_pressed')

func simulate_ui_down_event() -> void:
    var ui_down_event := InputEventAction.new()
    ui_down_event.action = 'ui_down'
    ui_down_event.pressed = true

    Input.parse_input_event(ui_down_event)

func _on_option_button_1_pressed() -> void:
    simulate_ui_down_event()

func _on_option_button_2_pressed() -> void:
    simulate_ui_down_event()
```

The issue I'm running into is the following: upon clicking an `OptionButton`, if I immediately press `ui_cancel` to hide the dropdown, **the next `ui_down` input is eaten entirely** (even if other inputs are made in between), and will therefore not change focus to the other `OptionButton`. Pressing `ui_down` a second time (and all subsequent times) will behave as expected. Note that this only happens if I immediately press `ui_cancel` after pressing an `OptionButton`; pressing `ui_down`/`ui_up` to navigate the options first and then pressing `ui_cancel` does not trigger the problem.

**Steps to reproduce:**
Load the reproduction project and press the first `OptionButton` (should have automatically grabbed focus) via `ui_enter`. Immediately press `ui_cancel` after the popup appears and then press `ui_down` and notice that the focus does not shift to the second `OptionButton`, even though it appears as though the first `OptionButton` has focus. Press `ui_down` again and notice that the focus now shifts properly to the second `OptionButton`.

**Minimal reproduction project:**
[parse-input-action-bug-reproduction-project.zip](https://github.com/godotengine/godot/files/5791470/parse-input-action-bug-reproduction-project.zip)

**Comment by Sauermann** [comment]
I am unable to replicate this issue (with a slightly adjusted MRP) in v4.1.beta3.official [ada712e06].
Seems to be restricted to 3.x.

**Comment by akien-mga** [comment]
It's a pretty old issue by now so I'll close as fixed in 4.1.

If anyone needs this fixed in 3.x and there's no duplicate report still open, we can reopen it for 3.x.
